src/server/RedirectingHttpHandler.ts

```diff
@@ -0,0 +1,70 @@
+import { getLoggerFor } from '../logging/LogUtil';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { HttpHandlerInput } from './HttpHandler';
+import { HttpHandler } from './HttpHandler';
+
+export class RedirectingHttpHandler extends HttpHandler {
+  private readonly logger = getLoggerFor(this);
+  private readonly matcher: RegExp;
+  private readonly redirects: { pattern: RegExp; redirect: string }[] = [];
+
+  /**
+   * Creates a handler for the provided redirects.
+   * @param redirects - A mapping between URL patterns.
+   */
+  public constructor(redirects: Record<string, string>, private readonly statusCode: number = 308) {
+    super();
+    const patterns = Object.keys(redirects);
+
+    // Create a single regexp for quick checks
+    this.matcher = new RegExp(`^${patterns.join('|')}$`, 'u');
+
+    // Create an array of (regexp, redirect) pairs
+    this.redirects = patterns.map(
+      (pattern): { pattern: RegExp; redirect: string } => ({
+        pattern: new RegExp(pattern, 'u'),
+        redirect: redirects[pattern],
+      }),
+    );
+  }
+
+  public async canHandle({ request }: HttpHandlerInput): Promise<void> {
+    // Only return if a redirect is configured for the requested URL
+    if (!this.matcher.test(request.url ?? '')) {
+      throw new NotImplementedHttpError(`No redirect configured for ${request.url}`);
+    }
+  }
+
+  public async handle({ request, response }: HttpHandlerInput): Promise<void> {
+    const { url } = request;
+
+    // Get groups and redirect of first matching pattern
+    const result = this.redirects.reduce<{ match: RegExpExecArray; redirect: string } | null>(
+      (prev, { pattern, redirect }): { match: RegExpExecArray; redirect: string } | null => {
+        if (prev) {
+          return prev;
+        }
+        const match = pattern.exec(url ?? '');
+        if (match) {
+          return { match, redirect };
+        }
+        return null;
+      },
+      null,
+    );
+
+    // Only return if a redirect is configured for the requested URL
+    if (!result) {
+      throw new NotImplementedHttpError(`No redirect configured for ${url}`);
+    }
+
+    // Build redirect URL from regexp result
+    const { match, redirect } = result;
+    const redirectUrl = match.reduce((prev, param, index): string => prev.replace(`$${index}`, param), redirect);
+
+    // Send redirect response
+    this.logger.info(`Redirecting ${url} to ${redirectUrl}`);
+    response.writeHead(this.statusCode, { location: redirectUrl });
```

Comment:
I wonder if we want to use the existing redirection mechanisms for this (they also come with headers, possible HTML responses, etc.)

Reply:
Where can I find those? Not sure what you are talking about...

Reply:
`RedirectHttpError` etc.

Reply:
Is this how you typically redirect? So I would then throw that error instead of writing the response?

Feels a bit weird to imagine redirection as an error...

Reply:
Yes—see it more as throwing an exception. Anything that interrupts the normal request flow can use it.
Although I am not sure about the gains in this situation so don't refactor it yet; pinging @joachimvh.

Reply:
In this case it might be ok not to do it. It's useful when there's a component somewhere deeper in that can throw a redirect so it gets caught by the error handler in the http handler and then converted into a correct response. But we're already at the root of the stack here, so in this case it would complicate the code since you would need to add an error handler and response writer.